src/cbadc/utilities.py

```diff
@@ -232,6 +232,54 @@ def read_byte_stream_from_url(
             pass
     raise StopIteration
 
+def csv_2_control_signal(filename: str, M: int, msb2lsb: bool = False):
+    """Creates an iterator that reads from a CSV file.  
+
+    Parameters
+    ----------
+    filename : `str`
+        filename for input file
+    M : `int`
+        number of controls
+    msb2lsb : `bool`
+        set order of control bits. Default is LSB to MSB
+
+    Yields
+    ------
+    array_like
+        a control signal sample
+
+    Example
+    -------
+    >>> # Given the following file, named `test.csv`, for a setup with four control bits (M=4):
```

Comment:
can we create this file in the same doctext example something like

```
with open('test.csv','w') as f:
    f.write("""
0,0,0,0
0,0,0,1
0,0,1,1
0,1,1,1
""")
```

takes as long as the comment above...

Reply:
ah good idea. I was wondering how to do this in a neat way.